unitary/alpha/quantum_world.py

```diff
@@ -305,6 +305,20 @@ def _interpret_result(self, result: Union[int, Iterable[int]]) -> int:
             return result_list[0]
         return result
 
+    def unhook(self, object: QuantumObject) -> None:
+        """Replaces all usages of the given object in the circuit with a new ancilla with value=0."""
+        # Creates a new ancilla.
+        new_ancilla = self._add_ancilla(object.name)
+        # Replace operations using the qubit of the given object with the new ancilla.
+        qubit_remapping_dict = {
+            object.qubit: new_ancilla.qubit,
+            new_ancilla.qubit: object.qubit,
+        }
+        self.circuit = self.circuit.transform_qubits(
+            lambda q: qubit_remapping_dict.get(q, q)
+        )
+        return
```

Comment:
The method works like reset.  (Maybe I should rename it to reset()?) So I don't think user should apply more effects on the ancilla. They should move forward with the reset `object`, which since now has a new reset value.